src/OpenSage.Game/Data/Apt/ConstantData.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenSage.Data.Utilities.Extensions;
+
+namespace OpenSage.Data.Apt
+{
+    public sealed class ConstantData
+    {
+        public List<ConstantEntry> Entries { get; internal set; }
+        public uint AptDataEntryOffset { get; internal set; }
+
+        public ConstantData()
+        {
+            Entries = new List<ConstantEntry>();
+        }
+
+        public static ConstantData FromFileSystemEntry(FileSystemEntry entry)
+        {
+            using (var stream = entry.Open())
+            using (var reader = new BinaryReader(stream))
+            {
+                var data = new ConstantData();
+
+                //validate that this is a correct const              
+                var magic = reader.ReadFixedLengthString(17);
+                if (magic!="Apt constant file")
+                    throw new InvalidDataException($"Not a supported const file: {magic}");
+
+                reader.BaseStream.Seek(3, SeekOrigin.Current);
+
+                data.AptDataEntryOffset = reader.ReadUInt32();
+                var numEntries = reader.ReadUInt32();
+                reader.BaseStream.Seek(4, SeekOrigin.Current);
+
+
+                for (var i=0;i< numEntries;++i)
```

Comment:
I'm not sure what spacing settings you're using here :) I should add an `.editorconfig` file for this stuff, so that Visual Studio will help with consistency.

Reply:
It is the default setting of VS2017